File: src/test/java/com/distribution/DistributionApplicationTests.java
package com.distribution;

import com.alibaba.fastjson.JSON;
import com.distribution.modules.api.pojo.vo.LoanOrderVO;
import com.distribution.modules.api.service.IdCardQueryService;
import com.distribution.modules.sys.service.SysUserService;
import com.google.common.collect.Maps;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.data.redis.core.RedisCallback;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.data.redis.serializer.RedisSerializer;
import org.springframework.http.MediaType;
import org.springframework.test.context.junit4.SpringRunner;
import org.springframework.test.web.servlet.MockMvc;
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
import org.springframework.test.web.servlet.setup.MockMvcBuilders;
import org.springframework.web.context.WebApplicationContext;

import java.io.IOException;

@RunWith(SpringRunner.class)
@SpringBootTest
public class DistributionApplicationTests {
    @Autowired
    SysUserService userService;
    @Autowired
    IdCardQueryService idCardQueryService;
    @Autowired
    WebApplicationContext context;
    @Autowired
    RedisTemplate<String, Object> redisTemplate;

    private MockMvc mockMvc;

    @Before
    public void setUp() throws Exception {
        mockMvc = MockMvcBuilders.webAppContextSetup(context).build();
    }

    @Test
    public void contextLoads() {
        userService.queryList(Maps.newHashMap()).forEach(System.out::println);

    }

    @Test
    public void testIdCard() throws IOException {
        System.out.println(redisTemplate.opsForValue().increment("worker_id", 1));
    }


    public long getIncrValue(final String key) {

        return redisTemplate.execute((RedisCallback<Long>) connection -> {
            RedisSerializer<String> serializer=redisTemplate.getStringSerializer();
            byte[] rowkey=serializer.serialize(key);
            byte[] rowval=connection.get(rowkey);
            try {
                String val=serializer.deserialize(rowval);
                return Long.parseLong(val);
            } catch (Exception e) {
                return 0L;
            }
        });
    }

    @Test
    public void testAop() throws Exception {
        mockMvc.perform(MockMvcRequestBuilders.get("/api/testDrools"))
                .andExpect(MockMvcResultMatchers.status().isOk()).andDo(MockMvcResultHandlers.print());
    }

    @Test
    public void testLoan() throws Exception {
        LoanOrderVO orderVO = new LoanOrderVO();
        orderVO.setOrderMobile("18635136850");
        orderVO.setOrderIdcardno("14052219860503301X");
        orderVO.setLoanId("18");
        mockMvc.perform(MockMvcRequestBuilders.post("/api/loan")
                .contentType(MediaType.APPLICATION_JSON_UTF8)
                .header("token","eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiI0ZDU2N2FjNjYwYzc0MWFlYTY1MzYzNjE3ZTcxMTk5YyIsImlhdCI6MTUzMTE4MjExMSwiZXhwIjoxNTMxODE1NzExfQ.p6kx0XTxXIj1I8DQHJ3Cr2Jy8ggy9DHUVxi1Vabz_nI")
                .content(JSON.toJSONString(orderVO)))
                .andExpect(MockMvcResultMatchers.status().isOk()).andDo(MockMvcResultHandlers.print());
    }

}

File: src/test/java/com/distribution/DistributionApplicationTests.kt
package com.distribution

import com.alibaba.fastjson.JSON
import com.distribution.modules.api.pojo.vo.LoanOrderVO
import com.distribution.modules.api.service.IdCardQueryService
import com.distribution.modules.dis.service.DisMemberInfoService
import com.distribution.modules.sys.service.SysConfigService
import com.distribution.modules.sys.service.SysUserService
import com.distribution.queue.LevelUpSender
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.data.redis.core.RedisTemplate
import org.springframework.http.MediaType
import org.springframework.test.context.junit4.SpringRunner
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.result.MockMvcResultHandlers
import org.springframework.test.web.servlet.result.MockMvcResultMatchers
import org.springframework.test.web.servlet.setup.MockMvcBuilders
import org.springframework.web.context.WebApplicationContext
import java.io.IOException

@RunWith(SpringRunner::class)
@SpringBootTest
class DistributionApplicationTests {
    @Autowired
    lateinit var userService: SysUserService
    @Autowired
    lateinit var context: WebApplicationContext
    @Autowired
    lateinit var redisTemplate: RedisTemplate<String, Any>
    @Autowired
    lateinit var idCardQueryService: IdCardQueryService
    @Autowired
    lateinit var disMemberInfoService: DisMemberInfoService
    @Autowired
    lateinit var sysConfigService: SysConfigService


    @Autowired
    lateinit var levelUpSender: LevelUpSender


    private lateinit var mockMvc: MockMvc

    @Before
    @Throws(Exception::class)
    fun setUp() {
        mockMvc = MockMvcBuilders.webAppContextSetup(context).build()
    }

    @Test
    fun contextLoads() {
        userService.queryList(mapOf()).forEach { println(it) }

    }

    @Test
    @Throws(IOException::class)
    fun testIdCard() {
        println(redisTemplate.opsForValue().increment("worker_id", 1))
    }

    @Test
    @Throws(Exception::class)
    fun testAop() {
        mockMvc.perform(MockMvcRequestBuilders.get("/api/testDrools"))
                .andExpect(MockMvcResultMatchers.status().isOk).andDo(MockMvcResultHandlers.print())
    }

    @Test
    @Throws(Exception::class)
    fun testLoan() {
        val orderVO = LoanOrderVO()
        orderVO.orderMobile = "18635136850"
        orderVO.orderIdcardno = "14052219860503301X"
        orderVO.loanId = "18"
        mockMvc.perform(MockMvcRequestBuilders.post("/api/loan")
                .contentType(MediaType.APPLICATION_JSON_UTF8)
                .header("token", "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiI0ZDU2N2FjNjYwYzc0MWFlYTY1MzYzNjE3ZTcxMTk5YyIsImlhdCI6MTUzMTE4MjExMSwiZXhwIjoxNTMxODE1NzExfQ.p6kx0XTxXIj1I8DQHJ3Cr2Jy8ggy9DHUVxi1Vabz_nI")
                .content(JSON.toJSONString(orderVO)))
                .andExpect(MockMvcResultMatchers.status().isOk).andDo(MockMvcResultHandlers.print())
    }

    @Test
    fun testIdCardQuery() {
        print("------------------------------------")
        print(idCardQueryService.isMatched("140109198701040532", "胡春亮"))
        print("------------------------------------")
    }


    @Test
    fun testCacheConfig() {
        //{"level_1":"568","level_2":"368","level_3":"168"}
        val value = "123"
        val key = "level_price"
        sysConfigService.updateValueByKey(key, value)
    }

    @Test
    fun testMyTeam() {
        //查询所有锁粉信息
        val myTeam = disMemberInfoService.findMyTeam("9f71511fad6d4e2d8b1f854e084a9d71")
        val disFansList = myTeam.filter { "0" == it.disUserType || it.disUserType == null }

        //所有代理信息
        val children = myTeam.filter { "1" == it.disUserType }

        //返回数据
        val map = mapOf("countFans" to disFansList.size, "fansList" to disFansList,
                "countChirldern" to children.size, "children" to children)

        print(map)
    }

}
